front/src/modules/ui/table/options/components/TableOptionsDropdownContent.tsx

```diff
@@ -124,31 +127,56 @@ export function TableOptionsDropdownContent({
     }
   }, [setViewEditMode]);
 
-  const handleViewNameSubmit = useCallback(
-    (event?: FormEvent) => {
-      event?.preventDefault();
+  const handleViewNameSubmit = useRecoilCallback(
+    ({ set, snapshot }) =>
+      async (event?: FormEvent) => {
+        event?.preventDefault();
 
-      if (viewEditMode.mode && viewEditInputRef.current?.value) {
-        const name = viewEditInputRef.current.value;
-        const nextViews =
-          viewEditMode.mode === 'create'
-            ? [...views, { id: v4(), name }]
-            : views.map((view) =>
-                view.id === viewEditMode.viewId ? { ...view, name } : view,
-              );
+        const name = viewEditInputRef.current?.value;
 
-        (onViewsChange ?? setViews)(nextViews);
-      }
+        if (!viewEditMode.mode || !name) {
+          return resetViewEditMode();
+        }
 
-      resetViewEditMode();
-    },
+        const views = await snapshot.getPromise(tableViewsState(tableScopeId));
+
+        if (viewEditMode.mode === 'create') {
+          const viewToCreate = { id: v4(), name };
+          const nextViews = [...views, viewToCreate];
```

Comment:
it was already there but nextViews is not clear to me, do we mean "new" (post creation) views? or "next in the list" views?

Reply:
I would prefer to call views: existingViews and nextViews: views

Reply:
`nextViews` here refers to the next state, in contrast to the current state (`views`). Would it be clearer if I name those: `currentViews` and `nextViews`?